app/client/components/accountoverview/contributionUpdateAmount.tsx

```diff
@@ -0,0 +1,89 @@
+import { css } from "@emotion/core";
+import { palette, space } from "@guardian/src-foundations";
+import { capitalize } from "lodash";
+import React, { Dispatch, SetStateAction, useState } from "react";
+import { formatDateStr } from "../../../shared/dates";
+import {
+  augmentInterval,
+  PaidSubscriptionPlan
+} from "../../../shared/productResponse";
+import { ProductType } from "../../../shared/productTypes";
+import { Button } from "../buttons";
+import { SuccessMessage } from "../delivery/address/deliveryAddressEditConfirmation";
+import { ProductDescriptionListTable } from "../productDescriptionListTable";
+
+import { ContributionUpdateAmountForm } from "./contributionUpdateAmountForm";
+
+interface ContributionUpdateAmountProps {
+  subscriptionId: string;
+  mainPlan: PaidSubscriptionPlan;
+  productType: ProductType;
+  nextPaymentDate: string | null;
+  amountUpdateStateChange: Dispatch<SetStateAction<number | null>>;
+}
+
+export const ContributionUpdateAmount = (
+  props: ContributionUpdateAmountProps
+) => {
+  enum Status {
+    OVERVIEW,
+    EDITING,
+    CONFIRMED
+  }
+
+  const [status, setStatus] = useState<Status>(Status.OVERVIEW);
```

Comment:
before this was done with a couple of bools, but I thought this may be a little clearer because those bools weren't allowed to both be true